### ISSUE TITLE
Cannot get into a game

### ISSUE THREAD

**Issue by Gruetzig** [kyechou/leagueoflegends#83]
### System information

 - Distro: Arch
 - DE/WM/Compositor: Gnome x11
 - CPU: intel i3-10100F
 - GPU: amd radeon r9 380
 - Graphics card drivers installed: mesa

> If your issue is related to graphics display, please attach the output of `vulkaninfo --summary`.
I do not know if it is, but I included vulkaninfo --summary in my pastebin.

### Describe the issue

When starting the game, not the launcher, it immediately crashes with an error I attached below, tried both ranked(bruh) and practice tool.

Reinstalled twice with no luck, worked just fine yesterday, never used the lutris version.

### Steps to reproduce

https://pastebin.com/BS7yBiru pastebin with vulkaninfo --summary and leagueoflegends -v start

1. Run command `leagueoflegends -v start`
2. Go into a game
3. See critical error 

### Screenshots

![image](https://user-images.githubusercontent.com/88926852/230069865-f9a854ce-cab2-4c97-bdb9-ffc281908eff.png)

**Comment by ExistingProgrammer** [comment]
Check out these threads:
https://www.reddit.com/r/leagueoflinux/comments/12c8w4p/last_patch_did_it_again/
https://www.reddit.com/r/leagueoflinux/comments/12cbqeq/update_of_leagueoflegendsgit_script_and_the_aur/

**Comment by ademblon** [comment]
I just installed kyechou version. (I'm usually using lutris, but I need a working version of lol asap, for a tournament)

I also have the same issues, and many others too as you can see in the second and third link you sent.

My error is 
`0328:err:seh:dispatch_exception unknown exception (code=c0000420) raised`

I also have the error 
`X11DRV_vkCreateWin32SurfaceKHR Failed to allocate client window` but I don't know if it's related. This one appears when starting the client, not when starting a game.

**Comment by Gruetzig** [comment]
Its nice to see that Im not alone, but why should I check these out? I dont see a solution there, please link a specific comment if I overlooked it.

**Comment by ceselder** [comment]
Exactly the same issue

**Comment by mohad12211** [comment]
same problem, fresh aur install, version: 13.7.1.r0.gfb60c87-1

**Comment by kyechou** [comment]
@Gruetzig, what's the package version of leagueoflegends-git were you using? Have you tried reinstall league?

**Comment by kyechou** [comment]
@Gruetzig , @ademblon , @ceselder Also could you run `leagueoflegends run which wine` and share the output?

**Comment by pierrepas** [comment]
I have the same issue even after a fresh install, and the same output for "run which wine" than @ceselder .
Get the critical error whenever joining a LoL or TFT game, and was working fine yesterday, so it is due to the 13.7 update.

**Comment by salova322** [comment]
@pierrepas  yea same

**Comment by Gruetzig** [comment]
> @Gruetzig, what's the package version of leagueoflegends-git were you using? Have you tried reinstall league?

local/leagueoflegends-git 13.7.1.r0.gfb60c87-1

**Comment by Gruetzig** [comment]
I will try reinstalling once more, wine is also in /opt/wine-lol/bin/wine

**Comment by Gruetzig** [comment]
reinstalled, same issue, https://pastebin.com/z8AQdbcT log if you need

**Comment by Bqleine** [comment]
I'm also experiencing this issue after doing a fresh install with [wine-lol 7.0.5](https://github.com/GloriousEggroll/wine-ge-custom/releases/tag/7.0-GE-5-LoL) manually.

**Comment by foxyllakora** [comment]
Here is the error I'm getting when the game tries to load in after champ select:
` 02f0:err:seh:dispatch_exception unknown exception (code=c0000420) raised`

That's the only error that shows up using `leagueoflegends -v start`

OS: EndeavourOS
CPU: Ryzen 9 5950x
GPU: AMD XFX 6950XT
Mobo: ASUS TUF Gaming X570-Plus
Ram: G.Skill DDR4 32GB

**Comment by FlashOnFire** [comment]
Same error as above
Arch Linux 6.2.9-arch1-1
Intel Core i5-11400
Nvidia RTX 3060 TI

`09ec:err:seh:dispatch_exception unknown exception (code=c0000420) raised`

**Comment by ademblon** [comment]
> ![image](https://user-images.githubusercontent.com/94973748/230119970-a07083d6-1058-4e74-a819-2f49104517fa.png)

Exact same for me

**Comment by Yuanhao-JIANG** [comment]
same issue here,
OS: EndeavourOS Linux x86_64
Kernel: 6.2.9-arch1-1 
Resolution: 3840x2160 
DE: Plasma 5.27.3 
WM: kwin
CPU: 11th Gen Intel i7-11700K (16) @ 4.900GHz 
GPU: NVIDIA GeForce RTX 3080 Ti

**Comment by romainaury** [comment]
Same here , happens as well with lutris installer wine-ge-lol-5.0-5
Also, I just reinstalled manjaro here. Went from kernel 5.15.104-2 to 6.1.8-1.
It was working perfectly fine before reinstalling/upgrading.
Don't really know if that could be a problem :/

OS: Manjaro 22.1.0 Talos
Kernel: x86_64 Linux 6.1.21-1-MANJARO
DE: GNOME 43.5
WM: Mutter
CPU: AMD Ryzen 5 2600 Six-Core @ 12x 3.4GHz
GPU: NVIDIA GeForce RTX 3060 Ti

**Comment by kyechou** [comment]
Hi all, thanks for sharing the system and logging information!

Yes, I can reproduce the error as of this morning. And as mentioned by previous comments and logs, this exception is where the game executable fails.
```
err:seh:dispatch_exception unknown exception (code=c0000420) raised
```

The issue is confirmed. And it'd be nice to also have people who are more familiar with Wine than I am look into it.

**Comment by PalanixYT** [comment]
It's not worth looking into because it's probably a problem on [Riot's end](https://www.reddit.com/r/leagueoflegends/comments/12c7qkl/comment/jf0v3b4/?context=3) and WINE just so happens to be able to trigger it perfectly

**Comment by kyechou** [comment]
Thanks @PalanixYT! That's good to know.

**Comment by noah1510** [comment]
> Hi all, thanks for sharing the system and logging information!
> 
> Yes, I can reproduce the error as of this morning. And as mentioned by previous comments and logs, this exception is where the game executable fails.
> 
> ```
> err:seh:dispatch_exception unknown exception (code=c0000420) raised
> ```
> 
> The issue is confirmed. And it'd be nice to also have people who are more familiar with Wine than I am looking into it.

I can confirm this behavior using my wrapper script on kubuntu 22.10 (See #82) using lutris-wine-ge-7.0-5.

**Comment by coolst3r** [comment]
same here on fedora

**Comment by trodz6** [comment]
same on void:
OS: Void Linux x86_64 
Kernel: 6.1.21_1 
AMD ATI Radeon RX 7900 XT
AMD Ryzen 5 7600X (12)
Throwing this in there. I was concerned it was due to my new hardware. Maybe this will make this issue searchable with similar hardware. :)

**Comment by SandstormCG** [comment]
I've been getting the same error `0158:err:seh:dispatch_exception unknown exception (code=c0000420) raised`. But what happens just before is that it fails because the amdvlk switchable graphics is the wrong elf class (64bits), then after the error it prints the call stack and unloads amdvlk. And then it prints the seh error. 

`DRIVER:           Found ICD manifest file /usr/share/vulkan/icd.d/amd_icd32.json, version "1.0.0"
DEBUG | DRIVER:   Searching for ICD drivers named /usr/lib32/amdvlk32.so
DRIVER:           Found ICD manifest file /usr/share/vulkan/icd.d/amd_icd64.json, version "1.0.0"
DEBUG | DRIVER:   Searching for ICD drivers named /usr/lib/amdvlk64.so
INFO:             /usr/lib/amdvlk64.so: wrong ELF class: ELFCLASS64
DRIVER:           Requested layer /usr/lib/amdvlk64.so was wrong bit-type. Ignoring this JSON
DEBUG | LAYER:    Loading layer library /usr/lib32/amdvlk32.so
INFO | LAYER:     Insert instance layer "VK_LAYER_AMD_switchable_graphics_32" (/usr/lib32/amdvlk32.so)
INFO:             /usr/lib/amdvlk64.so: wrong ELF class: ELFCLASS64
INFO | LAYER:     Requested layer "VK_LAYER_AMD_switchable_graphics_64" was wrong bit-type.
LAYER:            vkCreateInstance layer callstack setup to:
LAYER:               <Application>
LAYER:                 ||
LAYER:               <Loader>
LAYER:                 ||
LAYER:               VK_LAYER_AMD_switchable_graphics_32
LAYER:                       Type: Implicit
LAYER:                           Disable Env Var:  DISABLE_LAYER_AMD_SWITCHABLE_GRAPHICS_1
LAYER:                       Manifest: /usr/share/vulkan/implicit_layer.d/amd_icd32.json
LAYER:                       Library:  /usr/lib32/amdvlk32.so
LAYER:                 ||
LAYER:               <Drivers>
INFO | DRIVER:    linux_read_sorted_physical_devices:
INFO | DRIVER:         Original order:
INFO | DRIVER:               [0] AMD Radeon RX 6700 XT
INFO | DRIVER:         Sorted order:
INFO | DRIVER:               [0] AMD Radeon RX 6700 XT  
INFO | DRIVER:    linux_read_sorted_physical_devices:
INFO | DRIVER:         Original order:
INFO | DRIVER:               [0] AMD Radeon RX 6700 XT
INFO | DRIVER:         Sorted order:
INFO | DRIVER:               [0] AMD Radeon RX 6700 XT  
DEBUG | DRIVER:   Copying old device 0 into new device 0
DEBUG | LAYER:    Unloading layer library /usr/lib32/amdvlk32.so
0158:err:seh:dispatch_exception unknown exception (code=c0000420) raised
0204:err:ole:CoUninitialize Mismatched CoUninitialize
`

I don't know enough about wine , so I'm still trying to figure out why is getting this error as it seems is related to try to execute the 64 bit version of amdvlk when expecting 32 bit.

**Comment by coolst3r** [comment]
tryed new release on fedora 
+ shift
+ :
+ case "${1-}" in
+ break
+ case "${1-}" in
++ usage
++ cat
+++ basename ./package.sh
+ die 'Unknown distro: \n[!] Usage: package.sh [-h] [-v] <distro>
    Distro: all arch debian'
+ echo -e '[!] Unknown distro: \n[!] Usage: package.sh [-h] [-v] <distro>
    Distro: all arch debian'
[!] Unknown distro: 
[!] Usage: package.sh [-h] [-v] <distro>
    Distro: all arch debian
+ exit 1

**Comment by memand** [comment]
Here's the mem dump if anyone would find that useful..

https://kulturforeningenkasimir.dk/nextcloud/s/LoAJox6cdra4idM

**Comment by BlankSpruce** [comment]
OS: Manjaro
DE: KDE
CPU: i5-8300H
GPU: GTX 1650
Graphics cad drivers: nvidia

Logs collected with `WINEDEBUG=+seh+relay+tid` related to `League of Legends.exe`:
```
0a44:fixme:heap:EnumSystemFirmwareTables (0x4649524d, 0000000000000000, 0)
0a44:err:seh:dispatch_exception unknown exception (code=c0000420) raised
0a44:fixme:process:NtQueryInformationProcess ProcessCookie (0xffffffffffffffff,0x180b30650,0x00000004,(nil)) stub
0a44:fixme:thread:NtSetInformationThread info class 2 not supported yet
0a44:fixme:imm:ImeSetActiveContext (0x3003d0, 1): stub
0a44:fixme:imm:ImmReleaseContext (0000000000020114, 00000000003003D0): stub
```

The error code (0xc0000420) is related to [assertion failure](https://learn.microsoft.com/en-us/openspecs/windows_protocols/ms-erref/596a1078-e883-4972-9bbc-49e60bebca55) Given that:
- there were no known issues with all previously working Wine versions on LOL 13.6
- there are reports from Windows users on subreddit regarding "critical error"

it's reasonable to assume that the assertion failure comes due to some recent changes in LOL code. It just so happens that in Wine environment it's easier to fail that mysterious assertion. I'm looking forward to next couple of days since it's possible that there will be some patch to LOL itself to mitigate issue for Windows users at least.

**Comment by arnaudelub** [comment]
```bash
OS: Manjaro Linux x86_64 
██████████████████  ████████   Host: MS-7B18 1.0 
████████            ████████   Kernel: 5.15.104-2-MANJARO 
████████  ████████  ████████   Uptime: 1 hour, 33 mins 
████████  ████████  ████████   Packages: 1539 (pacman), 5 (flatpak), 3 (snap) 
████████  ████████  ████████   Shell: zsh 5.9 
████████  ████████  ████████   Resolution: 1920x1200, 3840x1600, 1920x1080 
████████  ████████  ████████   DE: GNOME 43.4 
████████  ████████  ████████   WM: Mutter 
████████  ████████  ████████   WM Theme: CustomAccentColors 
████████  ████████  ████████   Theme: Adw-dark [GTK2/3] 
████████  ████████  ████████   Icons: Papirus-Dark-Maia [GTK2/3] 
                               Terminal: terminator 
                               CPU: Intel i9-9900K (16) @ 5.000GHz 
                               GPU: NVIDIA GeForce RTX 2070 
                               Memory: 2079MiB / 64244MiB 

                                                       
```
same here, with the exact same error.
Using lutris with Wine version Lutris-ge-lol-7.0.5-x86_64

**Comment by SandstormCG** [comment]
Looking through google with the info of this reports [83#issue-1655469575](https://github.com/kyechou/leagueoflegends/issues/83#issue-1655469575) and [83#issuecomment-1498137251](https://github.com/kyechou/leagueoflegends/issues/83#issuecomment-1498137251)  it might be an error we should report on vulkan or wine repo, but i'm not sure if that's the case and if it is in which one, does anyone have more info?

**Comment by leobortolini** [comment]
Same issue here

OS: Zorin OS 16.2 x86_64
Kernel: 5.15.0-69-generic
CPU: Intel i7-9750H (12) @ 4.500GHz 
GPU: NVIDIA GeForce GTX 1660 Ti Mobile
wine-8.0
lutris-ge-lol-7.14-1-x86_64

**Comment by sekwah41** [comment]
From what I can see it seems to be happening to everyone, is there anyone that this isn't actually occurring to? Just mentioning this to make sure there isn't a flood of people just saying "doesn't work" with duplicate info.

**Comment by dykensp** [comment]
same error through Heroic game launcher + Wine-7.0-GE-5-LoL, was working until todays small lol update

**Comment by ceselder** [comment]
> From what I can see it seems to be happening to everyone, is there anyone or does anyone know anyone that this isn't occurring to? Just mentioning this to make sure there isn't a flood of people just saying "doesn't work" with duplicate info.

I'm 99% sure its working for literally noone on linux

**Comment by ExistingProgrammer** [comment]
> > From what I can see it seems to be happening to everyone, is there anyone or does anyone know anyone that this isn't occurring to? Just mentioning this to make sure there isn't a flood of people just saying "doesn't work" with duplicate info.
> 
> I'm 99% sure its working for literally noone on linux

Yup, and it is [crashing on windows](https://www.reddit.com/r/leagueoflegends/comments/12c7qkl/comment/jf0v3b4/?context=3) too with the exact same error. We just have to wait till riot fixes it.

**Comment by BlankSpruce** [comment]
Riot seems to be aware of the problem: https://old.reddit.com/r/leagueoflegends/comments/12c7qkl/patch_137_bug_megathread/jf1wufh/

**Comment by laqyx** [comment]
DRIVER | LAYER:          Using "AMD Radeon RX 5700 XT 50th Anniversary (RADV NAVI10)" with driver: "/usr/lib32/libvulkan_radeon.so"
06e0:err:vulkan:X11DRV_vkCreateWin32SurfaceKHR Failed to allocate client window for hwnd=0x200f0
0310:err:seh:dispatch_exception unknown exception (code=c0000420) raised

sadly crashing here too

**Comment by coolst3r** [comment]
[https://ibb.co/VMDjGhL](url)

**Comment by FlashOnFire** [comment]
Any updates on this ? 
Did someone get it working ?

**Comment by coolst3r** [comment]
@

> I've been getting the same error `0158:err:seh:dispatch_exception unknown exception (code=c0000420) raised`. But what happens just before is that it fails because the amdvlk switchable graphics is the wrong elf class (64bits), then after the error it prints the call stack and unloads amdvlk. And then it prints the seh error: [log paste](https://pastebin.com/GNZyq6zy)
> 
> I don't know enough about wine , so I'm still trying to figure out why is it getting this error, seems like is trying to execute the 64 bit version of amdvlk when expecting 32 bit. And is the same error as the original post of the issue just one is using libvulkan_radeon and the other amdvlk64

he error message you have shared suggests that an unknown exception has been raised with code "c0000420" while

**Comment by coolst3r** [comment]
> I've been getting the same error `0158:err:seh:dispatch_exception unknown exception (code=c0000420) raised`. But what happens just before is that it fails because the amdvlk switchable graphics is the wrong elf class (64bits), then after the error it prints the call stack and unloads amdvlk. And then it prints the seh error: [log paste](https://pastebin.com/GNZyq6zy)
> 
> I don't know enough about wine , so I'm still trying to figure out why is it getting this error, seems like is trying to execute the 64 bit version of amdvlk when expecting 32 bit. And is the same error as the original post of the issue just one is using libvulkan_radeon and the other amdvlk64

the error message indicates that an exception with code "c0000420" was raised and not handled by the SEH mechanism. The SEH mechanism is used by the Windows operating system to manage exceptions and errors in programs. However, without more context, it's difficult to determine the exact cause of this error. It could be related to a problem with Wine, the application being run, or the system environment.

**Comment by coolst3r** [comment]
> DRIVER | LAYER: Using "AMD Radeon RX 5700 XT 50th Anniversary (RADV NAVI10)" with driver: "/usr/lib32/libvulkan_radeon.so" 06e0:err:vulkan:X11DRV_vkCreateWin32SurfaceKHR Failed to allocate client window for hwnd=0x200f0 0310:err:seh:dispatch_exception unknown exception (code=c0000420) raised
> 
> sadly crashing here too

The error message suggests that there is an issue with creating a client window for the Vulkan-based graphics API. Specifically, it seems that there was a failure to allocate a client window for the specified window handle. Additionally, there is a separate error related to an unknown exception being raised with code c0000420.

**Comment by coolst3r** [comment]
> Logs collected with `WINEDEBUG=+seh+relay+tid` related to `League of Legends.exe`:
> 
> ```
> 0a44:fixme:heap:EnumSystemFirmwareTables (0x4649524d, 0000000000000000, 0)
> 0a44:err:seh:dispatch_exception unknown exception (code=c0000420) raised
> 0a44:fixme:process:NtQueryInformationProcess ProcessCookie (0xffffffffffffffff,0x180b30650,0x00000004,(nil)) stub
> 0a44:fixme:thread:NtSetInformationThread info class 2 not supported yet
> 0a44:fixme:imm:ImeSetActiveContext (0x3003d0, 1): stub
> 0a44:fixme:imm:ImmReleaseContext (0000000000020114, 00000000003003D0): stub
> ```

This log appears to be generated by a Windows application or process. It indicates that an unknown exception with error code c0000420 was raised, which could potentially indicate a critical system error. Additionally, there are several fixme messages which suggest that certain features or functions are not yet fully implemented or supported in the current version of the software.

The first line of the log, "EnumSystemFirmwareTables", appears to be related to enumerating firmware tables in the system, although it is unclear what specific action or purpose triggered this call. The subsequent lines, including the exception and fixme messages, suggest that the application or process may be encountering issues related to memory management, thread information, and input method editor (IME) context activation and release.

Without further context or analysis of the software and system environment in which this log was generated, it is difficult to determine the root cause of these errors and their potential impact on the system. However, it is clear that there are several areas of the software that require further development and debugging to ensure stable and reliable operation.

**Comment by coolst3r** [comment]
> I just installed kyechou version. (I'm usually using lutris, but I need a working version of lol asap, for a tournament)
> 
> I also have the same issues, and many others too as you can see in the second and third link you sent.
> 
> My error is `0328:err:seh:dispatch_exception unknown exception (code=c0000420) raised`
> 
> I also have the error `X11DRV_vkCreateWin32SurfaceKHR Failed to allocate client window` but I don't know if it's related. This one appears when starting the client, not when starting a game.

The first error is related to Vulkan, a graphics API. The error message indicates that the function "vkCreateWin32SurfaceKHR" failed to allocate a client window for the specified window handle. This error could occur due to a variety of reasons such as incorrect usage of Vulkan functions or issues with the window system.

The second error is related to structured exception handling (SEH), which is a mechanism for handling errors in Windows programs. The error message indicates that an unknown exception with error code "c0000420" was raised, which could indicate an access violation or other type of memory error.

Without more context or information, it is difficult to determine the exact cause of these errors. However, potential solutions could involve checking the code that uses Vulkan functions for errors or debugging memory issues in the program using a tool such as a debugger.

**Comment by PalanixYT** [comment]
Like I and @BlankSpruce have mentioned before, it's a problem on Riot's end.

**Comment by coolst3r** [comment]
> Like I and @BlankSpruce have mentioned before, it's a problem on Riot's end.

X11DRV_vkCreate "Win32"  SurfaceKHR

win 32  is for 32 bit which they just depracted

**Comment by PalanixYT** [comment]
> X11DRV_vkCreate "Win32" SurfaceKHR
> win 32 is for 32 bit which they just depracted
https://old.reddit.com/r/leagueoflegends/comments/12c7qkl/patch_137_bug_megathread/jf1wufh/

**Comment by coolst3r** [comment]
> > X11DRV_vkCreate "Win32" SurfaceKHR
> > win 32 is for 32 bit which they just depracted
> 
> That may be the case but it's still a problem on Riot's end https://old.reddit.com/r/leagueoflegends/comments/12c7qkl/patch_137_bug_megathread/jf1wufh/

yeah they dont support 32 bit anymore i read it hopefully they rollback the update

**Comment by zexa** [comment]
There's an interesting post that might help in the league of legends reddit: https://old.reddit.com/r/leagueoflegends/comments/12det14/found_a_fix_for_the_critical_error_crash_so_far/

**Comment by santiagosantos08** [comment]
> There's an interesting post that might help in the league of legends reddit: https://old.reddit.com/r/leagueoflegends/comments/12det14/found_a_fix_for_the_critical_error_crash_so_far/

That doesn't fix the issue at all. In Windows you can just get around the "critical mistake" by force quitting the client and reconnecting a couple times in a row which is done via task manager, but OP's game was getting stuck on top of it (because of Win full screen optimizations) so he couldn't kill League, the "fix" in question just solves that issue, the game is still crashing.

**Comment by fobekyo** [comment]
It seems to be related. If you enable window mode, while you still can't play a game, wine no longer displays a crash message.

**Comment by BlankSpruce** [comment]
> It seems to be related. If you enable window mode, while you still can't play a game, wine no longer displays a crash message. Edit: Sometimes it still does.

I've checked that and while it doesn't make game playable in any manner seemingly it MIGHT HAVE affected something. In particular, not all `League of Legends.exe` starts lead to known error. Some either immediately terminate application (no "critical error" dialog) or leave process in zombie state that needs to be manually killed. Out of couple tries I've collected these various logs:
```
0534:fixme:heap:EnumSystemFirmwareTables (0x4649524d, 0000000000000000, 0)
0534:err:seh:dispatch_exception unknown exception (code=c0000420) raised
0534:fixme:process:NtQueryInformationProcess ProcessCookie (0xffffffffffffffff,0x180b01650,0x00000004,(nil)) stub
0534:err:seh:call_stack_handlers invalid frame 0000000000116BA9 (0000000000022000-0000000000120000)
0534:err:seh:NtRaiseException Exception frame is not in stack limits => unable to dispatch exception.
```

```
11fc:fixme:heap:EnumSystemFirmwareTables (0x4649524d, 0000000000000000, 0)
11fc:err:seh:dispatch_exception unknown exception (code=c0000420) raised
11fc:fixme:process:NtQueryInformationProcess ProcessCookie (0xffffffffffffffff,0x180b01650,0x00000004,(nil)) stub
11fc:err:virtual:virtual_setup_exception stack overflow 2384 bytes in thread 11fc addr 0x17005a229 stack 0x206b0 (0x20000-0x21000-0x120000)
```

```
0880:fixme:heap:EnumSystemFirmwareTables (0x4649524d, 0000000000000000, 0)
0880:err:seh:dispatch_exception unknown exception (code=c0000420) raised
0880:fixme:process:NtQueryInformationProcess ProcessCookie (0xffffffffffffffff,0x180b01650,0x00000004,(nil)) stub
0880:err:virtual:virtual_setup_exception nested exception on signal stack in thread 0880 addr 0x7f9ff1e459b7 stack 0x67fef3a0
```

```
0b30:fixme:heap:EnumSystemFirmwareTables (0x4649524d, 0000000000000000, 0)
0b30:err:seh:dispatch_exception unknown exception (code=c0000420) raised
0b30:fixme:process:NtQueryInformationProcess ProcessCookie (0xffffffffffffffff,0x180b01650,0x00000004,(nil)) stub
0b30:err:seh:NtRaiseException Unhandled exception code c0000005 flags 0 addr 0x141294666
```

These point to stack overflow and/or segfault kind of errors.

We could eliminate the possibility that "Restore defaults" has any impact at all here if someone tried multiple tries to see if some of the behaviours above are seen also without "Restore defaults". I honestly don't think it should matter at all having in mind that reinstallation should have exact same effect but truthfully before "Restore defaults" I had 100% ratio of appeareance of "Critical error" dialogs and none of the above logs.

**Comment by coolst3r** [comment]
> It seems to be related. If you enable window mode, while you still can't play a game, wine no longer displays a crash message. Edit: Sometimes it still does.


vulkan isssue 

X11DRV_vkCreateWin32SurfaceKHR Failed to allocate client window
vulkan, a graphics API. The error message indicates that the function "vkCreateWin32SurfaceKHR" failed to allocate a client window for the specified window handle.

**Comment by BlankSpruce** [comment]
Interesting discovery. When I run with `WINEDEBUG=warn+seh` game doesn't crash and actually load to the beginning to immediately freeze visually - background music can be heard for instance:
![image](https://user-images.githubusercontent.com/32396809/230638839-0e00ba43-0cb7-4922-b46d-95df3dd1e26e.png)
Game doesn't start paused, it just happened after I've killed it the first time. Logs show tons of access violation errors around the log with `code=c0000420`:
```
05d0:warn:seh:dispatch_exception EXCEPTION_ACCESS_VIOLATION exception (code=c0000005) raised
05d0:warn:seh:dispatch_exception EXCEPTION_ACCESS_VIOLATION exception (code=c0000005) raised
05d0:warn:seh:dispatch_exception EXCEPTION_ACCESS_VIOLATION exception (code=c0000005) raised
05d0:warn:seh:dispatch_exception EXCEPTION_ACCESS_VIOLATION exception (code=c0000005) raised
05e4:err:seh:dispatch_exception unknown exception (code=c0000420) raised
05d0:warn:seh:dispatch_exception EXCEPTION_ACCESS_VIOLATION exception (code=c0000005) raised
05d0:warn:seh:dispatch_exception EXCEPTION_BREAKPOINT exception (code=80000003) raised
05d0:warn:seh:dispatch_exception EXCEPTION_ACCESS_VIOLATION exception (code=c0000005) raised
05d0:warn:seh:dispatch_exception EXCEPTION_ACCESS_VIOLATION exception (code=c0000005) raised
05d0:warn:seh:dispatch_exception EXCEPTION_ACCESS_VIOLATION exception (code=c0000005) raised
05d0:warn:seh:dispatch_exception EXCEPTION_ACCESS_VIOLATION exception (code=c0000005) raised
05d0:warn:seh:dispatch_exception EXCEPTION_ACCESS_VIOLATION exception (code=c0000005) raised
05d0:warn:seh:dispatch_exception EXCEPTION_ACCESS_VIOLATION exception (code=c0000005) raised
05d0:warn:seh:dispatch_exception EXCEPTION_ACCESS_VIOLATION exception (code=c0000005) raised
```
Game isn't playable at all but that's at least some progress.

**Comment by mk-giga** [comment]
> Game isn't playable at all but that's at least some progress.
> 
> EDIT: Actually it doesn't freeze immediately always, on some runs I have around 1 to 2 seconds of rendering. Since game is paused I only see this progress bar rendering.

So I've been having the same exception thrown as everyone else with `code=0000420` so I thought I'd run the game with `WINEDEBUG=warn+seh` as well, and this is what I get. (running this repos installer script)

```
0b4c:err:sync:RtlpWaitForCriticalSection section 000000017006B660 "../wine-lol/dlls/ntdll/loader.c: loader_section" wait timed out in thread 0b4c, blocked by 0b40, retrying (60 sec)
0b50:err:sync:RtlpWaitForCriticalSection section 000000017006B660 "../wine-lol/dlls/ntdll/loader.c: loader_section" wait timed out in thread 0b50, blocked by 0b40, retrying (60 sec)
0b54:err:sync:RtlpWaitForCriticalSection section 000000017006B660 "../wine-lol/dlls/ntdll/loader.c: loader_section" wait timed out in thread 0b54, blocked by 0b40, retrying (60 sec)
0b58:err:sync:RtlpWaitForCriticalSection section 000000017006B660 "../wine-lol/dlls/ntdll/loader.c: loader_section" wait timed out in thread 0b58, blocked by 0b40, retrying (60 sec)
0b5c:err:sync:RtlpWaitForCriticalSection section 000000017006B660 "../wine-lol/dlls/ntdll/loader.c: loader_section" wait timed out in thread 0b5c, blocked by 0b40, retrying (60 sec)
```
I know we're way past the point of realizing that memory allocation/reading is going wrong somewhere but I hope this information helps whoever's at the forefront of debugging this mess.

Specs:
OS: Arch Linux
CPU: Intel Core i7-6700k
GPU: NVIDIA GeForce 980 Ti
GPU Driver: Proprietary

**Comment by noah1510** [comment]
How did you get the game to pause? When I tried inputting anything the game completly crashed.

**Comment by polkaulfield** [comment]
> > Game isn't playable at all but that's at least some progress.
> > EDIT: Actually it doesn't freeze immediately always, on some runs I have around 1 to 2 seconds of rendering. Since game is paused I only see this progress bar rendering.
> 
> So I've been having the same exception thrown as everyone else with `code=0000420` so I thought I'd run the game with `WINEDEBUG=warn+seh` as well, and this is what I get. (running this repos installer script)
> 
> ```
> 0b4c:err:sync:RtlpWaitForCriticalSection section 000000017006B660 "../wine-lol/dlls/ntdll/loader.c: loader_section" wait timed out in thread 0b4c, blocked by 0b40, retrying (60 sec)
> 0b50:err:sync:RtlpWaitForCriticalSection section 000000017006B660 "../wine-lol/dlls/ntdll/loader.c: loader_section" wait timed out in thread 0b50, blocked by 0b40, retrying (60 sec)
> 0b54:err:sync:RtlpWaitForCriticalSection section 000000017006B660 "../wine-lol/dlls/ntdll/loader.c: loader_section" wait timed out in thread 0b54, blocked by 0b40, retrying (60 sec)
> 0b58:err:sync:RtlpWaitForCriticalSection section 000000017006B660 "../wine-lol/dlls/ntdll/loader.c: loader_section" wait timed out in thread 0b58, blocked by 0b40, retrying (60 sec)
> 0b5c:err:sync:RtlpWaitForCriticalSection section 000000017006B660 "../wine-lol/dlls/ntdll/loader.c: loader_section" wait timed out in thread 0b5c, blocked by 0b40, retrying (60 sec)
> ```
> 
> I know we're way past the point of realizing that memory allocation/reading is going wrong somewhere but I hope this information helps whoever's at the forefront of debugging this mess.
> 
> Specs: OS: Arch Linux CPU: Intel Core i7-6700k GPU: NVIDIA GeForce 980 Ti GPU Driver: Proprietary

Found this codeweavers post where a dev fixes an almost exactly identical bug (seh exception in dlls/ntdll/loader.c):
https://www.codeweavers.com/blog/aeikum+retired/2022/6/3/debugging-wine-with-x86-hardware-debug-registers

Ill try to look into it but for someone familiar with wine development this may help a lot

**Comment by dconz13** [comment]
Which patches are applied to wine staging 7.05 for league to work? Maybe we can get around the issue by applying similar patches to the latest version of wine staging?

**Comment by kyechou** [comment]
@dconz13 https://github.com/GloriousEggroll/wine-ge-custom/tree/67235b2be693a2274405253602c59e4f800ef003

**Comment by ceselder** [comment]
> @dconz13 The patches are here: https://github.com/GloriousEggroll/wine-ge-custom/tree/67235b2be693a2274405253602c59e4f800ef003

Thank god for GloriousEggRoll

**Comment by ceselder** [comment]
Still can't get into game? Reinstalling...

**Comment by polkaulfield** [comment]
These are the old patches, I'm trying the one from the post (It isn't applied to upstream AFAIK)
```diff
--- a/dlls/ntdll/path.c
+++ b/dlls/ntdll/path.c
@@ -519,7 +519,7 @@ static ULONG get_full_path_helper(LPCWSTR name, LPWSTR buffer, ULONG size)

     RtlAcquirePebLock();

-    if (NtCurrentTeb()->Tib.SubSystemTib)  /* FIXME: hack */
+    if (0 && NtCurrentTeb()->Tib.SubSystemTib)  /* FIXME: hack */
         cd = &((WIN16_SUBSYSTEM_TIB *)NtCurrentTeb()->Tib.SubSystemTib)->curdir.DosPath;
     else
         cd = &NtCurrentTeb()->Peb->ProcessParameters->CurrentDirectory.DosPath;

```

**Comment by kyechou** [comment]
@ceselder I was answering the question from @dconz13. The game still doesn't work.

**Comment by ceselder** [comment]
> @ceselder I was answering the question from @dconz13. The game still doesn't work.

Oh my bad. ):

**Comment by noah1510** [comment]
>  if (0 && NtCurrentTeb()->Tib.SubSystemTib) 

Doesn't this just disable the if part 🤔

**Comment by ceselder** [comment]
> > if (0 && NtCurrentTeb()->Tib.SubSystemTib)
> 
> Doesn't this just disable the if part thinking

yes

**Comment by BlankSpruce** [comment]
> How did you get the game to pause? When I tried inputting anything the game completly crashed.

I believe this is the feature of Practice Tool where if application terminates (crash or killed, doesn't matter) after the game has been "started" (as in clock starts ticking) the server will pause the game. In my case first non-crash lead to non-yet-paused game. After game has visually frozen I've killed `League of Legends.exe` and retried again. On second (and any consecutive try within the session) I'd see what I've posted above.

**Comment by polkaulfield** [comment]
Update! 

> These are the old patches, I'm trying the one from the post (It isn't applied to upstream AFAIK)
> 
> ```diff
> --- a/dlls/ntdll/path.c
> +++ b/dlls/ntdll/path.c
> @@ -519,7 +519,7 @@ static ULONG get_full_path_helper(LPCWSTR name, LPWSTR buffer, ULONG size)
> 
>      RtlAcquirePebLock();
> 
> -    if (NtCurrentTeb()->Tib.SubSystemTib)  /* FIXME: hack */
> +    if (0 && NtCurrentTeb()->Tib.SubSystemTib)  /* FIXME: hack */
>          cd = &((WIN16_SUBSYSTEM_TIB *)NtCurrentTeb()->Tib.SubSystemTib)->curdir.DosPath;
>      else
>          cd = &NtCurrentTeb()->Peb->ProcessParameters->CurrentDirectory.DosPath;
> ```
> 
> Its still compiling, I'll update when I check if this fixes anything.

This patch makes the critical error window disappear. Game just silently crashes.

Output of `leagueoflegends -v start`

```
INFO | DRIVER:               [0] AMD Radeon RX 580 Series (RADV POLARIS10)  
DEBUG | DRIVER:   Copying old device 0 into new device 0
09e8:err:seh:dispatch_exception unknown exception (code=c0000420) raised
```

The (code=c0000420) error is the only one that remains when not adding any debug flag. Seems that theres something else to patch.

@kyechou told me that some logs pointed to `/dlls/ntdll/unix/signal_x68_64.c`

The patch we use now instead of the custom glibc is iirc this one:
[LoL-abi.vsyscall32-alternative_patch_by_using_a_fake_cs_segment.patch](https://github.com/GloriousEggroll/wine-ge-custom/blob/7.0-GE-LoL/patches/wine-hotfixes/LoL/LoL-abi.vsyscall32-alternative_patch_by_using_a_fake_cs_segment.patch)

Reading through the code I have noticed that it only applies to `/dlls/ntdll/unix/signal_i386.c`, that makes sense, game was 64bit before.

We need someone to port that patch to `/dlls/ntdll/unix/signal_x68_64.c` and check if it fixes the issue. At least the critical error window bug is gone now...

**Comment by BlankSpruce** [comment]
> 

I don't want to kill the mood but the devil is in the details. Unless you tried running `League of Legends.exe` under winedbg to confirm that code in `dlls/ntdll/loader.c` is in fact reason for assertion failure we don't have strong reasons to believe that issue described in your link is related to our problem. 

Additionally to be somewhat more confident on correlation you'd have to make sure that `tid` (first four characters before `:err`) of log:
```
0b4c:err:sync:RtlpWaitForCriticalSection section 000000017006B660 "../wine-lol/dlls/ntdll/loader.c: loader_section" wait timed out in thread 0b4c, blocked by 0b40, retrying (60 sec)
```
matches that of `dispatch_exception unknown exception`.

**Comment by polkaulfield** [comment]
@BlankSpruce You are right, I'm getting the same behavior with that codeweavers patch. It was just luck that i didnt get the window the first run. Anyways I managed to run `League of Legends.exe` with winedbg.
Here's the output:
```
WineDbg starting on pid 012c
0x0000000180350f71 stub+0x350f71: ret
Wine-dbg>c
0130:warn:seh:dispatch_exception EXCEPTION_ACCESS_VIOLATION exception (code=c0000005) raised
0x0000000170058735: ret
Wine-dbg>c
0130:warn:seh:dispatch_exception EXCEPTION_PRIV_INSTRUCTION exception (code=c0000096) raised
Unhandled exception: privileged instruction in 64-bit code (0x0000000145345040).
Register dump:
 rip:0000000145345040 rsp:000000000011fe38 rbp:0000000000000000 eflags:00010202 (  R- --  I   - - - )
 rax:000000017008a050 rbx:0000000000000000 rcx:0000000067ff0000 rdx:0000000145345000
 rsi:0000000000000000 rdi:0000000000000000  r8:0000000067ff0000  r9:00000000ffffffff r10:0000000000000000
 r11:0000000000000000 r12:0000000000000000 r13:0000000000000000 r14:0000000000000000 r15:0000000000000000
Stack dump:
0x000000000011fe38:  000000007b62c669 0000000000000000
0x000000000011fe48:  0000000000000000 0000000000000000
0x000000000011fe58:  0000000000000000 0000000000000000
0x000000000011fe68:  0000000170060093 0000000000000000
0x000000000011fe78:  0000000000000000 0000000000000000
0x000000000011fe88:  0000000000000000 0000000145345000
0x000000000011fe98:  0000000067ff0000 ffffffffffffffff
0x000000000011fea8:  000000017006b440 00000001700259c0
0x000000000011feb8:  0000000000000000 000000000011fea0
0x000000000011fec8:  0000000000000000 000000000011fe70
0x000000000011fed8:  0000000000000000 0000000000000000
0x000000000011fee8:  0000000000000000 0000000000000000
Backtrace:
=>0 0x0000000145345040 in league of legends (+0x5345040) (0x0000000000000000)
  1 0x000000007b62c669 ActivateActCtx+0x20ae9() in kernel32 (0x0000000000000000)
0x0000000145345040 league of legends+0x5345040: inb     %dx,%al
```

This `ActivateActCtx` function seems related...

**Comment by kyechou** [comment]
> @BlankSpruce You are right, I'm getting the same behavior with that codeweavers patch. It was just luck that i didnt get the window the first run. Anyways I managed to run `League of Legends.exe` with winedbg. Here's the output:
> 
> ...
> 
> This `ActivateActCtx` function seems related...

Is it possible to continue the execution or hit the `ERROR_ASSERTION_FAILURE` (0xC0000420) exception?

**Comment by polkaulfield** [comment]
Couldnt manage to do it sadly

**Comment by Luka-J9** [comment]
I think something might have changed. I just downloaded a patch and when attempting to connect I didn't get the message box of doom. But I am still not able to connect to the game

**Comment by Laci0503** [comment]
> ```
> Backtrace:
> =>0 0x0000000145345040 in league of legends (+0x5345040) (0x0000000000000000)
>   1 0x000000007b62c669 ActivateActCtx+0x20ae9() in kernel32 (0x0000000000000000)
> 0x0000000145345040 league of legends+0x5345040: inb     %dx,%al
> ```

I'n expert on wine, but isn't 'inb' a privilaged instruction that can only be run in kernelspace?
I've never looked at how exactly wine works, but this caught my eye in the logs.

**Comment by LucianoBatista** [comment]
Seems that this new build of GE solve the problem, but I was not able to build on my own.
https://github.com/GloriousEggroll/wine-ge-custom/tree/7.0-GE-LoL

**Comment by fobekyo** [comment]
are you sure that it solves the problem? I still can't get into a game even after updating to https://github.com/GloriousEggroll/wine/tree/ge-lol-7.0.7

**Comment by ceselder** [comment]
> are you sure that it solves the problem? I still can't get into a game even after updating to https://github.com/GloriousEggroll/wine/tree/ge-lol-7.0.7

Its not fixed for many people yeah. @kyechou can get into games though. I think it only works for him so far?

**Comment by LucianoBatista** [comment]
I was not able to test it. I'm justing passing the information I got from a Reddit thread.

https://www.reddit.com/r/leagueoflinux/comments/12cei7t/patch_137_feedback_megathread_transition_to_64bit/?utm_source=share&utm_medium=android_app&utm_name=androidcss&utm_term=1&utm_content=share_button

I was hopping for this to work 😕

**Comment by codrinsah** [comment]
> Its not fixed for many people yeah. @kyechou can get into games though. I think it only works for him so far?

I wasn't able to find any particular release for ge-lol-7.0.7 yet, so I presume he managed to build from that branch locally and make it work.

**Comment by ExistingProgrammer** [comment]
> > Its not fixed for many people yeah. @kyechou can get into games though. I think it only works for him so far?
> 
> I wasn't able to find any particular release for ge-lol-7.0.7 yet, so I presume he managed to build from that branch locally and make it work.

Yup. He submitted a [pull request](https://github.com/GloriousEggroll/wine-ge-custom/commit/5220e4fe6b3a6e59909e8348fdb1283301173886) to fix the error.
This script will probably be updated then to fix issue here as well?

**Comment by santiagosantos08** [comment]
> This script will probably be updated then to fix issue here as well?

If the fix is already merged to wine-lol then you can build it yourself or wait for a binary release (takes a couple hours). This script doesn't take care of the installation by itself, it just checks if you are running wine from the right directory "opt/wine-lol" instead of the usual "/usr/bin".

The update will depend on whichever method you used to get wine-lol in the first place (M-reimer's AUR pkg, deb package, etc)

**Comment by kyechou** [comment]
Hi all, the PR did *NOT* solve all the issues. Please don't run it just yet.. (or at your own risk)

**Comment by PalanixYT** [comment]
@kyechou Could you post your PKGBUILD and launcher script?

**Comment by kyechou** [comment]
@PalanixYT I need to do some testing first. I was building wine out of my local dir. I'll probably post my pkgbuild later today if I get a chance.

**Comment by PalanixYT** [comment]
I've heard something about a launcher script. Can you post that?

**Comment by kyechou** [comment]
It's already updated, but again, it's not a solution.

**Comment by kyechou** [comment]
Please also check out the update comment: https://github.com/GloriousEggroll/wine-ge-custom/pull/9#issuecomment-1500896901

**Comment by PalanixYT** [comment]
Oh sorry, I didn't catch that. I've heard that you were using a custom script. There is also a person on reddit claiming you've made a binary for Arch Linux.
https://www.reddit.com/r/leagueoflinux/comments/12cei7t/comment/jffnqhr/?utm_source=share&utm_medium=web2x&context=3
Anyways, thank you for your help

**Comment by kyechou** [comment]
@PalanixYT  I just found a typo in the last commit. please pull again if you haven't done so.

**Comment by kyechou** [comment]
### Status update
If we run the game with `WINEDEBUG=fixme-all,trace+seh,trace+vulkan,trace+window`, we can enter and play a match, which was tested through practice tool and verified by multiple people. But without the tracing flags, a new exception occurs:
```
err:virtual:virtual_setup_exception stack overflow 2000 bytes in thread 0ab0 addr 0x1700349b0 stack
```
Note that the address `0x1700349b0` is the same reported from different people. We assume there might be data race between the wine threads causing the stack overflow exception.

At the moment it seems the PR (https://github.com/GloriousEggroll/wine-ge-custom/pull/9) does not make any difference except for the error messages produced. Either way, eventually someone (or Riot) would need to investigate the cause of the stack overflow exception.

### Notes
Setting the debug tracing flags is not a fix but a temporary workaround for those who are addicted to the (wonderful) game, and it **doesn't seem to work for everyone**, so please try a few practice tool or AI matches first. For those who are interested, please follow [this guide](https://www.reddit.com/r/leagueoflinux/comments/12foijk/crashing_issue_temporary_fix_arch/) and proceed at your own risk.


Finally. Please comment on this issue only when there are **actionable information or log messages**. Thanks ❤

**Comment by kyechou** [comment]
@user6304 which wine version/patch set were you using? If it's from AUR then it's GE 7.0.5 without the patches. You may want to try the patched version following guide linked above, but still someone needs to fix it properly.

**Comment by kyechou** [comment]
[Note to self and others who're looking into the issue]

With the new patches and without the tracing flags, sometimes it shows:
```
0c08:err:seh:call_stack_handlers invalid frame 000000000013FC70 (0000000000022000-0000000000120000)
0c08:err:seh:NtRaiseException Exception frame is not in stack limits => unable to dispatch exception.
```
The exact error messages seem non-deterministic and dependent on individual setups and the wine compile-time flags. But so far they are all related to how the stack/frame pointers were manipulated. I haven't touched x86 or x64 asm for a long time, so it'd be helpful if anyone's interested in looking into this.

**Comment by PalanixYT** [comment]
've noticed that using the new leagueoflegends wrapper script I can join matches using 7.0.5.

**Comment by Definite007** [comment]
> 've noticed that using the new leagueoflegends wrapper script I can join matches using 7.0.5.
but are you able to play full match?
If yes then share the script

**Comment by mewkl** [comment]
Can someone link a binary/(beta)release of this latest wine version so I can link it to my lutris install?
I've never compiled wine before and this would be amazing help. 

I could not manage to compile yesterday, kept getting the same error about it not finding the wine src directory, but it's kind of hard for me to figure it out since it seems to be inside a virtual machine and then again inside a container.

**Comment by KJ2ME** [comment]
I am using lutris-ge-lol-7.0-5 with Bottles adding WINEDEBUG=trace+seh to Environment Variables and this is my results:

- With trainign tools farm 10mins on line all is fine, then farm jg including 1 dragon another 10min, then kill baron and destroy nexus with a lot of minios of the 3 lines (baron buff active) witthout problems.
- 3 ranked games played without problems.

Now with only test purposes and no other intentions I will play a lot of games :).

If someone need anything more from with my tests, let me know.

**Comment by kyechou** [comment]
Reminder:
> Please comment on this issue only when there are actionable information or log messages.

This is not a support thread.

**Comment by fattyre** [comment]
moonshadow565 seems to have patched wine in a way to make the game work or on reddit. I linked his post and the AUR link to the wine version. I'm downloading and trying now.

https://www.reddit.com/r/leagueoflinux/comments/12gkfiu/patches_for_64bit_update/

https://aur.archlinux.org/packages/wine-lol-staging

**Comment by kyechou** [comment]
Thanks for the information @fattyre! Unfortunately it still doesn't seem to work for everyone. It worked for some though.
(ref: https://github.com/GloriousEggroll/wine-ge-custom/pull/9)

**Comment by fattyre** [comment]
Ok so i tested everything out and seems to work well. I get this error in the log as I play though.

wine client error:688: write: Bad file descriptor
wine client error:690: write: Bad file descriptor
wine client error:548: write: Bad file descriptor
wine client error:734: write: Bad file descriptor
wine client error:bc0: write: Bad file descriptor
wine client error:bb8: wine client error:b08: wine client error:124: write: Bad file descriptor

Is this of any use or concern?

**Comment by kyechou** [comment]
@fattyre I have no idea... but it works it works :)
Just curious. What's your CPU/GPU? Are you on amd?

**Comment by fattyre** [comment]
> @fattyre I have no idea... but if it works it works :) Just curious. What's your CPU/GPU? Are you on amd?
> 
> Oh and by the way, did you disable the WINEDEBUG flags?

I am on EndeavourOS
CPU: AMD Ryzen 9 5900HS
GPU: NVIDIA GeForce RTX 3060 Mobile / Max-Q

**Comment by laqyx** [comment]
without the winedebug options and wine-lol-staging it always takes exacty 27-28 seconds for my game to crash
with the winedebug options and yesterdays gdrive wine build it takes everywhere from 10s-1m for my game to crash
heres what I could capture https://pastes.io/jufolsnvjy

**Comment by coolst3r** [comment]
i tryed make install fedora does nothing

**Comment by mk-giga** [comment]
i just realized something and i am fairly certain now,
since people are crashing -after- some time with an access violation in ram, it is most likely that some function in a dll is attempting to write to a buffer past the maximum buffer length in a 32 bit array of some sort. i remember vaguely some people having a message that showed some value slightly past the maximum 32bit integer limit

**Comment by kyechou** [comment]
Updated log: https://pastebin.com/65kwU7sn
Explanation:
* exception code `c0000005` is access violation. In this case that is a virtual page fault in wine triggered by SIGSEGV in linux.
* `info_0` in the log is the virtual page fault type. `0` is read fault. `1` is write fault. and `8` is execute fault.
* `stack_ptr` is the aligned rsp at the time.

I've narrowed down the cause to this:

* Stack overflow is caused by a strange unhandled virtual page fault.
* Most page faults in the log seem benign, except for the one occurring at `0x1700608de`. The exact address may differ depending on the game exe versions. Note that this address may be different from the one reported by the stack overflow error message.
* The page fault wasn't handled properly and so the `segv_handler` kept getting trigger, eventually overrun the stack space.

**Comment by kyechou** [comment]
Unfortunately I cannot continue working on this issue for a few weeks. For debugging and testing purposes, I've pushed my changes to [this branch](https://github.com/kyechou/wine-ge-custom/tree/8.5-testing). To print out the page fault debug messages like the ones in the log file, after running `patches/protonprep-LoL.sh`, you need to manually apply the [`debug.patch`](https://github.com/kyechou/wine-ge-custom/blob/8.5-testing/debug.patch).

**Comment by mk-giga** [comment]
Additional helpful information discovered:
The time of crash is dependent on CPU clock frequency.

Supporting evidence:

- I set my CPU's clock speed to 4.2Ghz and went into a custom game as the same champion 4 times, only standing still in spawn every time.

- I crash at 16 minutes every time.

- I now set my CPU's clock speed to 2.1Ghz and go into a custom game 4 times. I now crash at 32 seconds every time.

Further speculation:

 The data race theory some of you guys had has to be true at this point.  I also noticed that the address `0x1700608de` in error messages in kyechou's logs point to a binary value of just barely higher than 32 bits, so I am 100% convinced at this point that some 32-bit data buffer is being written to as if it's a 64-bit data structure, which, I mean, makes sense considering this crash is happening on a patch where 32-bit systems got deprecated.

If anyone here is familiar with the WINAPI and wine debugging process + willing to hold my hand a little, I'd love to sign away most of my free time to help fix this problem

**Comment by EndChapter** [comment]
Hi I am crawling into wine pull requests about this bug and i found [this pull request](https://gitlab.winehq.org/wine/wine/-/merge_requests/1907) makes a lot of sense getting access violation. Can you try with this patch? Thanks.

**Comment by kyechou** [comment]
@mk-giga Thanks for the observation! But there are a few things that don't seem right.

> The data race theory some of you guys had has to be true at this point.

If the root cause is related to a data race, I doubt the crash timing will be this deterministic. Typically the timing should be fairly unpredictable (especially on the same machine). IMHO the predictable timing behavior that you observed further indicates that the faster CPU got to the problematic instruction faster where it triggered the page faults twice as fast and also overran the stack twice as fast. So at the moment I don't think the current issue involves much about race conditions.

> I also noticed that the address 0x1700608de in error messages in kyechou's logs point to a binary value of just barely higher than 32 bits (it would require at least 33 bits to actually store)

If you look closely in the log, there are a lot of other addresses where the exception occurred that also require 33 bits (i.e., 64-bit locations). But they didn't seem to cause any trouble and simply triggered a one-time page fault, which is very normal for a program to do. So if the problem was a 32-bit data buffer being written, it doesn't make sense that a ton of other similar page faults (at different code locations) that worked just fine.

Another thing is, if you take a look at the wine source code where the additional debugging output code was added. In that function there's this line `rec.ExceptionAddress = (void *)RIP_sig(ucontext);`, which means the addresses in the log were the rip values at the time of (right before) the page fault exceptions. So it's definitely pointing to a code segment (i.e., text section) rather than a data buffer.

> If anyone here is familiar with the WINAPI and wine debugging process + willing to hold my hand a little, I'd love to sign away most of my free time to help fix this problem

I'd suggest taking a look at eggroll's wine-ge-custom repository, getting familiar with applying the patches and compiling the wine code, see how to reproduce the problems, and then you can look deeper into what's happening in the wine code, try tweaking a few things and see what happens.

**Comment by kyechou** [comment]
> Hi I am crawling into wine pull requests about this bug and i found [this pull request](https://gitlab.winehq.org/wine/wine/-/merge_requests/1907) makes a lot of sense getting access violation. Can you try with this patch? Thanks.

@EndChapter Thanks! That's a good find! It does seem likely to be related. I'll test it in the background when I get a chance, but no ETA guaranteed.

**Comment by kyechou** [comment]
@EndChapter Unfortunately there's the same error after applying the patch. I guess the next step would be to somehow attach winedbg to the address where the strange page faults happened and see what the instruction actually tried to do. Namely, figure out what/where the instruction at the exception address tried to read since it's a read fault.

**Comment by fvalasiad** [comment]
So, basically, there is a stack overflow error, in rito's code, it happens both in linux & windows, but for some reason in linux using wine it happens instantly, while windows users experience it after playing for a certain amount of time.

Adding debug flags to wine's runtime, effectively adding overhead to the program, makes the exception appear later in the course of the game, depending on your overall system(CPU, etc).

A race condition sounds like a good guess, since as it seems the error is highly dependent on the overall system used. But you said that you aren't sure about it, since the time it takes to throw the error is rather predictable, but i beg to differ.

The first time i entered a practice tool, it took 6 minutes to crash

the second, it took 2 minutes

third one crashed instantly

fourth one lasted 10 minutes.

## Conclusion
If we've concluded that it's a fault on rito's part, what are we even trying to find here?

**Comment by kyechou** [comment]
> If we've concluded that it's a fault on rito's part, what are we even trying to find here?

Because we don't know for sure if it's riot's fault or not. It's just speculation.

**Comment by kyechou** [comment]
It's good if we can confirm that of course.

**Comment by noah1510** [comment]
@fvalasiad If the problem is a data race, it is not a data race between CPU threads. If that were the case limiting wine to one thread would fix it however doing that only make the client perform terrible (yes even worse than normal). The fact that it runs with debug options enabled, means that in addition to riots problems, there is a problem with wine.

**Comment by fvalasiad** [comment]
@noah1510 Wdym limiting wine to one thread, if the problem lies in rito's code's threads anything we do in wine doesn't change anything. Do you mean that we configure wine to run all of the applications threads in a single thread? So that no two threads run in parallel?

In that case the race condition won't disappear by any means, it will just depend entirely on the scheduler's decisions.

Adding overhead to the program by enabling debug flags makes everything run slower, and as a result, the statistical chance that a race condition happens just gets smaller, thus the unpredictability.

Can you elaborate why you think that wine is at fault(potentially) too?

**Comment by fvalasiad** [comment]
@noah1510 race conditions aren't only occurring when threads run in parallel, they can occur anyways depending on the task switches of the scheduler.

You are correct though that limiting the parallelism, should also limit the chance that a race condition happens. So I understand your suspicion. 

I wouldn't trust that to pinpoint if wine is at fault tho, since task switching is rather complicated. Adding an extra `printf; fflush(stdout)` might as well make a race condition disappear(Given that it's extremely slow). Wine does just that when it prints debug logs to `stderr`(i assume).

Again, these are just predictions.

**Comment by kyechou** [comment]
Yes I agree that there are a lot things we don't know for sure yet. But we do know for a fact that, for those who have crashes, the crash (sooner or later) is caused by repeated page faults at a specific code location. This behavior is deterministic and reproducible. If we understand what the instruction is trying to do, we would know if it's fixable in Wine or if there's nothing we can do to circumvent that. I suggest if anyone has time, we should try to disassemble that part of code for a better understanding.

**Comment by fvalasiad** [comment]
@kyechou Are you suggesting we decompile the code to x86 assembly and try to debug it? ahahahha, isn't that against rito's terms of usage?

**Comment by kyechou** [comment]
@fvalasiad well.. i mean, what if we just attach the debugger while it's running? I may need to review their TOS though.

**Comment by fvalasiad** [comment]
@kyechou Tbh reading x86 assembly code that is the result of a compilation process(thus, even less purpose for it to be readable) is bad enough already, I'd rather not be reading machine code on top of that. Have we got any news regarding the windows side of things? Have they resolved the crashes there?

**Comment by kassindornelles** [comment]
@kyechou do you mind joining the linux gaming dev discord channel? we have dxvk and wine developers there 
would be interesting for you to share your findings so maybe we can figure this out faster on mainstream

feel free to share your findings in the # wine channel
https://discord.gg/linuxgamingdev

for others, please dont join if your purpose is to get user support, is a proton/wine/dxvk/vkd3d-proton development server and is not meant for that.

**Comment by moonshadow565** [comment]
Here is some info on ``0008-ntdll-nopguard-call_vectored_handlers.patch``:
Originally i assumed the game hooks the function midway based on return address but after inspecting generated wine code pre and post patch i observed following:

1. Current code generates 2 call sites for ``ret = func( &except_ptrs )`` in call_vectored_handlers
2. One of the call sites is only invoked on first run (i assume to initialize statics inside TRACE macro)
3. The game then proceeds to save return address after ``func`` invocation
4. On following invocations func returns from different call site
5. The game probably then detects this as modification of code and crashes intentionally?

I verified this behavior by simulating different call sites from 2 identical functions at random.
This is also why forcing tracing on seh made it past this check.

A cleaner patch that bunch of nops would probably be to force single call site by moving it into different function something like what is done with ``EXC_CallHandler`` in ``call_stack_handlers``.

**Comment by mewkl** [comment]
Hello, I was one of those people that could not play more after minions spawned in practice tool no matter if I was using "trace+seh", or "fixme-all,trace+seh,trace+vulkan,trace+window" or neither, even with the "wine-lol-staging-8.5.1-x86_64-64bitfix" runner I downloaded from reddit. 

After thinking a bit about race conditions I removed the WINEDEBUG environment variable and just set "restrict number of cores to" to "1" in lutris. (still interesting because it says core and not (logical) thread)
Game went on a lot longer, but still crashed after a while, so I tried again this time with the large WINEDEBUG env var and I'm pretty sure I can play a full game now, from practice tool testing. My pc has 40 threads btw. Maybe some1 can share this knowledge on the reddit post? I have no reddit account so I'm just posting this here.

**Comment by moonshadow565** [comment]
Here is some other tips for https://github.com/kyechou/wine-ge-custom/blob/388e786fcac9d79705c110996d41d1fd0ef3153a/patches/protonprep-LoL.sh#L36 

-  ``LoL-6.17+-syscall-fix.patch`` and ``LoL-abi.vsyscall32-alternative_patch_by_using_a_fake_cs_segment.patch`` and ``LoL-abi-vsyscall-fix.patch`` can be droped since game is not 32bit anymore
- ``LoL-seh-assertion-failure.patch`` won't apply  since wine-7.14 (079179fec720780679db3eed90adcc257f4ff240) since a similar patch seems to be upstreamed

Aside from that a pro tip is that you can test using a replay .rofl file or op.gg .bat file instead of going thru pains of launching client all the time.  Replay files are saved under Documents/League\ of\ Legends folders.
You can launch a replay or spectate stream like this
```
bin/wine cmd.exe /C 'start "" lol.exe EUW1-xxxxx.rofl'
```
Note: op.gg .bat files use ``start`` command to launch spectator streams. There is some difference between how start handles launching a process and how bare bones wine command launches a process.  I found out start to be necessary to launch the game properly for both rofl files and spectator stream (failing another check otherwise i assume).

**Comment by mewkl** [comment]
Hello guys, I just played an entire game without crashing with WINEDEBUG unset and Output debugging info disabled on the lutris runner options. 
All I did was disable hyperthreading on my bios. 
I am still using wine-lol-staging-8.5.1-x86_64-64bitfix and "Restrict number of cores = 1" with lutris. 
This might maybe be a much cleaner temporary fix until there is an actual patch out. 
The only downside is a slooow client/champ select. Game itself runs well on my ancient machine.

**Comment by kyechou** [comment]
Thanks @moonshadow565! I think what you suggested seem to fix the game exe crashing problem. I'll fix the patches and test more thoroughly.

**Comment by TheFeelTrain** [comment]
I think Riot may have pushed a hotfix, mine is suddenly working without me changing anything it might not be anything you did @kyechou

**Comment by fvalasiad** [comment]
I will test as well, gimme 2 mins

**Comment by fvalasiad** [comment]
I've now opened a practice tool and letting it sit afk hitting baron, will update in 20 minutes if it crashed or not. it would previously crash at less than 10 minutes.

**Comment by fvalasiad** [comment]
Game without `WINEDEBUG` crashes instantly, now testing with `trace+seh`

**Comment by TheFeelTrain** [comment]
I should have mentioned I have it working using the `wine-lol-staging` package from the AUR. Going back and trying `wine-lol-7.0-5` on the same prefix the riot client doesn't even open.

**Comment by kyechou** [comment]
@TheFeelTrain Yes I can confirm that it's working after I reverted back some patches.

**Comment by tridoxx** [comment]
> Game without `WINEDEBUG` crashes instantly, now testing with `trace+seh`
> 
> game with `trace+seh` crashes instantly, now testing with `fixme-all,trace+seh,trace+vulkan,trace+window`

use this wine version https://github.com/polkaulfield/lol-for-linux-installer/releases/download/wine-lol-staging/wine-lol-staging-8.5-1.tar.xz  

unzip it where are runners installed, for debian based is /home/<user_folder>/.local/share/lutris/runners/wine

**Comment by fvalasiad** [comment]
> > Game without `WINEDEBUG` crashes instantly, now testing with `trace+seh`
> > game with `trace+seh` crashes instantly, now testing with `fixme-all,trace+seh,trace+vulkan,trace+window`
> 
> use this wine version https://github.com/polkaulfield/lol-for-linux-installer/releases/download/wine-lol-staging/wine-lol-staging-8.5-1.tar.xz
> 
> unzip it where are runners installed, for debian based is /home/<user_folder>/.local/share/lutris/runners/wine

Hmmm is this wine staging with GE'S patches applied?

**Comment by kyechou** [comment]
It is. I think it uses a binary package built from the AUR package.

**Comment by PalanixYT** [comment]
It seems as though Riot Games has finally published a new patch?
I'm on leagueoflegends-git and using the precompiled binary by kyechou and I was able to play a match perfectly, without any crashes

**Comment by kyechou** [comment]
Thanks Riot for this one. I'll test a few older 32-bit patches (to make sure that no remaining 32-bit components still need them before removing them), and then update the public-facing stuff.

**Comment by fvalasiad** [comment]
I am at 17 minutes using `WINEDEBUG` but i am definitely trying the one you mentioned guys too, it looks good, guess league is back on linux

**Comment by PalanixYT** [comment]
Yup, I've tried multiple Practice Tools rounds and just finished a 45 minute game without any issues

**Comment by hurutparittya** [comment]
Can confirm, it seems to be working. Pog.

**Comment by fvalasiad** [comment]
Someone bother to notify the subreddit, so that mods post the solution, share the news with everyone else!

**Comment by fvalasiad** [comment]
Yeah GE's wine with WINEDEBUG now runs perfectly fine, i will now disable WINEDEBUG and try the other binary.

**Comment by kassindornelles** [comment]
That means only this bug remains upstream:
https://bugs.winehq.org/show_bug.cgi?id=54793

Please if anyone is interested at fixing this one on upstream wine you are more than welcome

**Comment by subcero123** [comment]
My client is not crashing but no window is displayed after joining a match.

**Comment by mateus-filho** [comment]
we have a solution guys? i lost

**Comment by polkaulfield** [comment]
> It is. I think it uses a binary package built from the AUR package.
It's just the binaries made with moonshadow's PKGBUILD
All credits to him. I just extracted the Arch package and compressed the binaries.
https://aur.archlinux.org/packages/wine-lol-staging

I uploaded it to work with https://github.com/polkaulfield/lol-for-linux-installer

It's just https://github.com/kassindornelles/lol-for-linux-installer

With the runner changed.
Kudos to moonshadow and kassidornelles :)

**Comment by polkaulfield** [comment]
Also with moonshadow's build it seems that the seh debug flag is not necessary, thats why i removed it from the script.

**Comment by kyechou** [comment]
Closing the issue now for being fixed by Riot.

Please be patient for a release on the fix. It takes time to compile/test to weed out some unneeded patches and there are some proton performance-related patches that were originally applied in the old GE 7.0.5 that may be worthwhile to rebase to 8.5. I'll send a PR to GE and M-Reimer to update the wine builds once it's done. In the meantime, please feel free to scroll up and try out other builds/installers.

For those interested in working on upstreaming the right fixes to mainstream wine, please check out https://bugs.winehq.org/show_bug.cgi?id=54793

**Comment by kassindornelles** [comment]
> @kyechou
> 
> > It is. I think it uses a binary package built from the AUR package.
> 
> It's just the binaries made with moonshadow's PKGBUILD All credits to him. I just extracted the Arch package and compressed the binaries. https://aur.archlinux.org/packages/wine-lol-staging
> 
> I uploaded it to work with https://github.com/polkaulfield/lol-for-linux-installer
> 
> It's just https://github.com/kassindornelles/lol-for-linux-installer
> 
> With the runner changed. Kudos to moonshadow and kassidornelles :)

do you plan in maintaining your fork of my installer?
i'm not sure if i want to come back to it anytime soon so would be nice to have someone to pass the torch

**Comment by kyechou** [comment]
Oh it's not my fork. I think @polkaulfield is maintaining the fork. @polkaulfield what do you think?

**Comment by moonshadow565** [comment]
> Closing the issue now for being fixed by Riot and @moonshadow565.
> 
> Please be patient for a release on the fix. It takes time to compile/test to weed out some unneeded patches and there are some proton performance-related patches that were originally applied in the old GE 7.0.5 that may be worthwhile to rebase to 8.5. I'll send a PR to GE and M-Reimer to update the wine builds once it's done. In the meantime, please feel free to scroll up and try out other builds/installers.
> 
> For those interested in working on upstreaming the right fixes to mainstream wine, please check out https://bugs.winehq.org/show_bug.cgi?id=54793

Some notes regarding ``LoL-client-slow-start-fix.patch``:
It's a very hacky workaround that was developed more or less by trial and error.
One reason why it maybe "works" better on GE-7.0 is possible differences between:
[LoL-launcher-client-connectivity-fix-0001-ws2_32-Return-a-valid-value-for-WSAIoctl-SIO_IDEAL_S.patch](https://github.com/GloriousEggroll/wine-ge-custom/blob/3e43da2c95dec5126a7d41025b606c777c50d381/patches/wine-hotfixes/LoL/LoL-launcher-client-connectivity-fix-0001-ws2_32-Return-a-valid-value-for-WSAIoctl-SIO_IDEAL_S.patch) 
and its usptream version in wine [493db3b1f799c9af1e3bad5fcf974aa0f3861cca](https://github.com/wine-mirror/wine/commit/493db3b1f799c9af1e3bad5fcf974aa0f3861cca) .
Hardcoded value of ``0x10000; /* 64k */`` standouts in both version of these patches.

**Comment by fvalasiad** [comment]
> > Closing the issue now for being fixed by Riot and @moonshadow565.
> > Please be patient for a release on the fix. It takes time to compile/test to weed out some unneeded patches and there are some proton performance-related patches that were originally applied in the old GE 7.0.5 that may be worthwhile to rebase to 8.5. I'll send a PR to GE and M-Reimer to update the wine builds once it's done. In the meantime, please feel free to scroll up and try out other builds/installers.
> > For those interested in working on upstreaming the right fixes to mainstream wine, please check out https://bugs.winehq.org/show_bug.cgi?id=54793
> 
> Some notes regarding `LoL-client-slow-start-fix.patch`: It's a very hacky workaround that was developed more or less by trial and error. One reason why it maybe "works" better on GE-7.0 is possible differences between: [LoL-launcher-client-connectivity-fix-0001-ws2_32-Return-a-valid-value-for-WSAIoctl-SIO_IDEAL_S.patch](https://github.com/GloriousEggroll/wine-ge-custom/blob/3e43da2c95dec5126a7d41025b606c777c50d381/patches/wine-hotfixes/LoL/LoL-launcher-client-connectivity-fix-0001-ws2_32-Return-a-valid-value-for-WSAIoctl-SIO_IDEAL_S.patch) and its usptream version in wine [493db3b1f799c9af1e3bad5fcf974aa0f3861cca](https://github.com/wine-mirror/wine/commit/493db3b1f799c9af1e3bad5fcf974aa0f3861cca) . Hardcoded value of `0x10000; /* 64k */` standouts in both version of these patches.

Also worth noting, https://github.com/GloriousEggroll/wine-ge-custom/blob/3e43da2c95dec5126a7d41025b606c777c50d381/patches/wine-hotfixes/LoL/LoL-launcher-client-connectivity-fix-0001-ws2_32-Return-a-valid-value-for-WSAIoctl-SIO_IDEAL_S.patch#L139

uses the index number 297, an index number which is used in the upstream version too https://github.com/wine-mirror/wine/blob/master/include/wine/afd.h#L286

**Comment by polkaulfield** [comment]
@kassindornelles 
> do you plan in maintaining your fork of my installer? i'm not sure if i want to come back to it anytime soon so would be nice to have someone to pass the torch

Sure!

**Comment by polkaulfield** [comment]
I plan to add a PBE version too, it's missing on lutris.

**Comment by subcero123** [comment]
Can someone explain me why the AUR package does not contains the new fixes?

**Comment by Niedzwiedzw** [comment]
```
04f8:err:vulkan:X11DRV_vkCreateWin32SurfaceKHR Failed to allocate client window for hwnd=0x1011a
0874:err:seh:dispatch_exception unknown exception (code=c0000420) raised
0878:err:virtual:virtual_setup_exception stack overflow 2256 bytes in thread 0878 addr 0x170058fa3 stack 0x1500730 (0x1500000-0x1501000-0x1600000)
087c:err:sync:RtlpWaitForCriticalSection section 000000017006B660 "?" wait timed out in thread 087c, blocked by 0000, retrying (60 sec)
0880:err:sync:RtlpWaitForCriticalSection section 000000017006B660 "?" wait timed out in thread 0880, blocked by 0000, retrying (60 sec)
0884:err:sync:RtlpWaitForCriticalSection section 000000017006B660 "?" wait timed out in thread 0884, blocked by 0000, retrying (60 sec)
0888:err:sync:RtlpWaitForCriticalSection section 000000017006B660 "?" wait timed out in thread 0888, blocked by 0000, retrying (60 sec)
088c:err:sync:RtlpWaitForCriticalSection section 000000017006B660 "?" wait timed out in thread 088c, blocked by 0000, retrying (60 sec)
```

this happens after champion select... I tried reinstalling everything everything and I'm using the master branch... any ideas friends? :(
